src/main/java/org/gridsuite/study/server/service/StudyService.java

```diff
@@ -462,6 +466,23 @@ public List<EquipmentInfos> searchEquipments(@NonNull UUID studyUuid, @NonNull U
         }
     }
 
+    private void cleanModifiedEquipments(List<EquipmentInfos> equipmentInfos) {
+        Set<EquipmentInfos> equipmentToDelete = new HashSet<>();
+        Map<String, List<EquipmentInfos>> groupedById = equipmentInfos.stream()
+                .collect(Collectors.groupingBy(EquipmentInfos::getId));
+
+        groupedById.forEach((id, equipments) -> {
+            if (equipments.size() > 1) {
+                equipmentToDelete.addAll(
+                        equipments.stream()
+                                .filter(e -> VariantManagerConstants.INITIAL_VARIANT_ID.equals(e.getVariantId()))
+                                .collect(Collectors.toList())
+                );
+            }
+        });
+        equipmentInfos.removeAll(equipmentToDelete);
```

Comment:
Why we don't delete directly in the block `forEach` ?

Reply:
We cannot delete elements from a list while iterating over it